rust/core/src/graph.rs

```diff
@@ -811,6 +811,55 @@ mod tests {
         graph.graph.return_sub_graph("test_model").unwrap();
     }
 
+    #[tokio::test]
+    async fn test_return_sub_graph_with_snapshot() {
+        let fs = FileSystem {
+            files: vec![
+                ("quary.yaml", "sqliteInMemory: {}"),
+                (
+                    "models/orders_snapshot_model.sql",
+                    "SELECT * from q.orders_snapshot",
+                ),
+                (
+                    "models/snapshots/orders_snapshot.snapshot.sql",
+                    "SELECT * FROM q.raw_orders",
+                ),
+                (
+                    "models/schema.yaml",
+                    r#"
+                snapshots:
+                - name: orders_snapshot
+                  unique_key: id
+                  strategy:
+                    timestamp:
+                      updated_at: updated_at
+                sources:
+                - name: raw_orders
+                  path: project_id.dataset_id.orders_snapshot
+                "#,
+                ),
+            ]
+            .iter()
+            .map(|(name, content)| {
+                (
+                    name.to_string(),
+                    quary_proto::File {
+                        name: name.to_string(),
+                        contents: Bytes::from(content.to_string()),
+                    },
+                )
+            })
+            .collect(),
+        };
+        let db = DatabaseQueryGeneratorSqlite::default();
+
+        let project = parse_project(&fs, &db, "").await.unwrap();
+
+        let graph = project_to_graph(project).unwrap();
+
+        graph.graph.return_sub_graph("orders_snapshot").unwrap();
```

Comment:
We probably want to assert a few things on this returned graph, e.g. that it returns the underlying snapshot?